### PR TITLE
Explore: Remove datasource testing on selector

### PR DESCRIPTION
**Pull request by davkal**
- datasource testing gets in the way of fast query iteration: switching
between datasources can take seconds
- it should not be explore's duty to test datasources in the first place
- removed the concept of datasourceError in Explore, should not be its
concern
- datasource errors will express themselves in query errors just fine
- failed language providers will show up as "(... not found)" in the cascade selectors
- disabled cascade buttons when language providers or meta queries come back empty
- connection errors are still bubbled up
- removed reconnection logic from explore, should not be its concern
- missing labels in loki are still "visible" via an empty label selector
- Loki and Prometheus treated connection errors differently than other
datasources, making sure to pass through the original error message

**Special notes for your reviewer**:
- I'd like to keep the "expensive" labels test for loki in the datasource config page itself, it just got in the way of using loki quickly in explore
- try switching between several Lokis, used to take ages

This is how a missing language support due to a failed connection is displayed now:
Screenshot 2019-10-28 at 17 42 05

### CODE REVIEW

**Review by hugohaggmark — CHANGES_REQUESTED** [review]
Great work but I found some cases where this is not working as expected:
1. Shouldn't Prometheus metrics button be disabled when we're unable to connect to data source?
2. Shouldn't Loki labels button be disabled when we're unable to connect to data source?
3. Shouldn't Influx/Logs measurements button be disabled and an error shown when we're unable to connect to data source? This one is particular user confusing as you can't do anything at all when Explore can't connect to the Influx data source.

Something that is out of scope for this PR but it would be cool if we could test the websocket connection when we test a Loki data source and not fail the test but warn the user.

File: public/app/plugins/datasource/prometheus/components/PromQueryField.tsx [hunk 1]
```diff
@@ -159,21 +156,6 @@ class PromQueryField extends React.PureComponent<PromQueryFieldProps, PromQueryF
...
-      return;
-    }
-
-    if (this.languageProviderInitializationPromise) {
-      this.languageProviderInitializationPromise.cancel();
-    }
-
-    if (this.languageProvider) {
-      this.refreshMetrics(makePromiseCancelable(this.languageProvider.fetchMetrics()));
-    }
```
[the remark this answers is not in the document]
Comment (davkal) [reply public/app/plugins/datasource/prometheus/components/PromQueryField.tsx]: Nope, this code path only triggered during a reconnect, which is no longer possible.

**Comment by davkal** [comment]
> Something that is out of scope for this PR but it would be cool if we could test the websocket connection when we test a Loki data source and not fail the test but warn the user.

done in #19555

**Review by hugohaggmark — APPROVED** [review]
LGTM! Works as expected and tested Influx, Loki and Prometheus. Nitpick: Thought the Loki label dropdown still looks enabled compared to Prometheus.

Loki:
image

Prometheus:
image

**Comment by aocenas** [comment]
@davkal Seems like this is fine but there are some issues with how things display:

Influx right away:
Screenshot 2019-11-11 at 13 44 35

Elastic metrics/logs has an error banner right away:
Screenshot 2019-11-11 at 13 44 49
Screenshot 2019-11-11 at 13 46 31

Loki has error only after you try to do a query.
Screenshot 2019-11-11 at 13 45 07

**Pull request merged**